CIScripts/Tests/TCPCommunicationTest.ps1

```diff
@@ -1,3 +1,67 @@
+           [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+    Write-Host "Running TCP Communication test"
+
+    Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+
+    $NetworkName = $TestConfiguration.DockerDriverConfiguration.NetworkConfiguration.NetworkName
+
+    Write-Host "Creating containers"
+    $ServerID, $ClientID = Invoke-Command -Session $Session -ScriptBlock {
+        $ServerID = docker run --network $Using:NetworkName -d iis-tcptest
+        $ClientID = docker run --network $Using:NetworkName -d microsoft/nanoserver ping -t localhost
+        return $ServerID, $ClientID
+    }
+
+    . $PSScriptRoot\CommonTestCode.ps1
+
+    Write-Host "Getting VM NetAdapter Information"
+    $VMNetInfo = Get-RemoteNetAdapterInformation -Session $Session -AdapterName $TestConfiguration.AdapterName
+
+    Write-Host "Getting Containers NetAdapter Information"
+    $ServerNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ServerID
+    $ClientNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ClientID
+
+    Write-Host "Configuring..."
+    Invoke-Command -Session $Session -ScriptBlock {
+        vif.exe --add $Using:VMNetInfo.IfName --mac $Using:VMNetInfo.MACAddress --vrf 0 --type physical
+        vif.exe --add HNSTransparent --mac $Using:VMNetInfo.MACAddress --vrf 0 --type vhost --xconnect $Using:VMNetInfo.IfName
+
+        vif.exe --add $Using:ServerNetInfo.AdapterShortName --mac $Using:ServerNetInfo.MACAddress --vrf 1 --type virtual --vif 1
+        vif.exe --add $Using:ClientNetInfo.AdapterShortName --mac $Using:ClientNetInfo.MACAddress --vrf 1 --type virtual --vif 2
+
+        nh.exe --create 1 --vrf 1 --type 2 --el2 --oif 1
+        nh.exe --create 2 --vrf 1 --type 2 --el2 --oif 2
+        nh.exe --create 3 --vrf 1 --type 6 --cen --cni 1 --cni 2
+
+        rt.exe -c -v 1 -f 1 -e ff:ff:ff:ff:ff:ff -n 3
+        rt.exe -c -v 1 -f 1 -e $Using:ServerNetInfo.MACAddress -n 1
+        rt.exe -c -v 1 -f 1 -e $Using:ClientNetInfo.MACAddress -n 2
+    }
+
+    Write-Host "Executing netsh"
+    Invoke-Command -Session $Session -ScriptBlock {
+        docker exec $Using:ServerID netsh interface ipv4 add neighbors $Using:ServerNetInfo.AdapterFullName $Using:ClientNetInfo.IPAddress $Using:ClientNetInfo.MACAddressWindows
+        docker exec $Using:ClientID netsh interface ipv4 add neighbors $Using:ClientNetInfo.AdapterFullName $Using:ServerNetInfo.IPAddress $Using:ServerNetInfo.MACAddressWindows
+    } | Out-Null
+
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        $ServerIP = $Using:ServerNetInfo.IPAddress
+        docker exec $ClientID powershell "Invoke-WebRequest -Uri http://${ServerIP}:8080/ -ErrorAction Continue" | Write-Host
+        return $LASTEXITCODE
+    }
+
+    Write-Host "Removing containers"
+    Invoke-Command -Session $Session -ScriptBlock { docker rm -f $Using:ServerID } | Out-Null
```

Comment:
I'm not sure, but removing containers without removing relevant vifs look like a dangerous, possibly crash-inducing thing. Maybe we should secure vRouter KM against this?